client/invoice_table.js

```diff
@@ -1,51 +1,84 @@
 TemplateController('invoice_table', {
   state: {
-    limit: 20
+    limit: 20,
+    searchTerms: {}
   },
 
   helpers: {
     invoices() {
-      return InvoiceTicketsCollection.find(getFilter(date.get(), days.get()), {
+      let filter = MongoDBQueryFactory.mergeDBFilters(MongoDBQueryFactory.createTimeFilter(date.get(), days.get()),MongoDBQueryFactory.createSearchFilter(Template.instance().state.searchTerms()));
+      return InvoiceTicketsCollection.find(filter, {
         sort: {
-          createdAt: sortState.get('sortCreatedAt', true),
-          total: sortState.get('sortTotal', true)
+          createdAt: Template.instance().sortingToDBSorting[StateMachine.get('sortCreatedAt', 'Asc')],
+          total: Template.instance().sortingToDBSorting[StateMachine.get('sortTotal', 'Asc')]
```

Comment:
Is there a reason why you are not using `this` to reference the `Template.instance()`?